src/Autofac/Builder/RegistrationBuilder.cs

```diff
@@ -149,7 +149,7 @@ public static class RegistrationBuilder
             Guid id,
             RegistrationData data,
             IInstanceActivator activator,
-            IEnumerable<Service> services)
+            Service[] services)
```

Comment:
These are definitely breaking changes but I don't think these methods would be used directly by consumers very often.

Reply:
Agreed. Pretty low risk.

Reply:
I brought all the changes from this file into the `develop` branch.